### PR TITLE
#176 create notification system

### PR DESCRIPTION
**Pull request by marek1123**
I created front-end and back-end part of notification system (without push notifications, it will be implemented in another task), I also created an example notification for successful exchange with simple constant text (this will also be extended in another task - show exchange request in Sidebar after clicking the notification).

@marektavac and @martingranak or @mdomdo0  can you please review back-end and front-end ?

### CODE REVIEW

**Comment by marek1123** [comment]
@marektavac and @martingranak or @mdomdo0 can you please review back-end and front-end ?

File: Backend/BlockChangesService.cs [hunk 1]
```diff
@@ -18,10 +18,15 @@ public BlockChangesService(IMongoDatabase database)
             _blockChangesCollection = database.GetCollection<BlockChangeRequest>(nameof(BlockChangeRequest));
         }
 
-        public async Task<bool> AddAndFindMatch(BlockChangeRequest entityToAdd)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="blockChangeRequest"></param>
+        /// <returns>bool value if corresponding second BlockChangeRequest was found and the found BlockChangeRequest</returns>
```
Comment (marektavac) [inline Backend/BlockChangesService.cs]: Cannot be solved just with one parameter and null means not found?

File: Backend/Interfaces/INotificationService.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public interface INotificationService
+    {
+        Task UpdateNotificationReadState(Guid notificationId, bool read);
+
+        /// <summary>
+        /// Get all notifications where user with studentId is recipient.
+        /// </summary>
+        /// <param name="studentId">Notification's recipientId.</param>
+        /// <returns></returns>
+        Task<IEnumerable<Notification>> GetStudentNotifications(Guid studentId);
```
Comment (marektavac) [inline Backend/Interfaces/INotificationService.cs]: You need all? If teher will be too much of  them?

File: WebAPI/Controllers/ExchangeController.cs [hunk 3]
```diff
@@ -16,24 +17,37 @@ namespace WebAPI.Controllers
...
+            currentUserBlockChangeRequest.BlockTo = BlockForExchangeModel.ConvertToBlock(request.BlockTo);
+            currentUserBlockChangeRequest.Status = ExchangeStatus.WaitingForExchange;
+            currentUserBlockChangeRequest.DateOfCreation = DateTime.Now;
+            currentUserBlockChangeRequest.StudentId = Guid.Parse(request.StudentId);
+
+            var (exchangeWasMade, foundMatch) = await _blockChangesService.AddAndFindMatch(currentUserBlockChangeRequest);
 
-            var res = await _blockChangesService.AddAndFindMatch(blockChangeRequest);
-            return Ok(res);
+            if (exchangeWasMade)
```
Comment (marektavac) [inline WebAPI/Controllers/ExchangeController.cs]: Controller should be responsible for generating notification? May be it should be the task for changeservice

**Pull request merged**